client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -275,10 +277,129 @@ define(['./module', 'angular', 'd3'], function (module, angular, d3) {
       }
     };
 
+    function validateYears(){
+      return $scope.params.objectives!==undefined &&
+             $scope.params.objectives.year!==undefined &&
+             $scope.params.objectives.year.start!==undefined &&
+             $scope.params.objectives.year.end !==undefined &&
+             ($scope.params.objectives.year.start < $scope.params.objectives.year.end);
```

Comment:
As i said, the whole controller should be refactored. Especially the validation methods because there are multiple functions that basically do the same checking over and over again. I'll add a new commit with some refactoring.